docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,984 @@
+
+## What is a source plugin?
+
+Source plugins "source" data from remote or local locations into what Gatsby calls [nodes](/docs/node-interface/). This tutorial uses a demo API so that you can see how the data works on both frontend and backend, but the same principles apply if you would like to source data from another API.
+
+For more background on source plugins, check out [Gatsby's source plugin documentation](/docs/creating-a-source-plugin/)
+
+## Why create a source plugin?
+
+Source plugins convert data from any source into a format that can be processed by Gatsby. Your Gatsby site could use several source plugins to combine data in interesting ways.
+
+There may not be [an existing plugin](/plugins/?=gatsby-source) for your data source, so you can create your own.
+
+_**NOTE:** if your data is local i.e. on your file system and part of your site's repo, then you generally don't want to create a new source plugin. Instead you want to use [gatsby-source-filesystem](/packages/gatsby-source-filesystem/) which handles reading and watching files for you. You can then use [transformer plugins](/plugins/?=gatsby-transformer) like [gatsby-transformer-yaml](/packages/gatsby-transformer-yaml/) to make queryable data from files._
+
+## How to create a source plugin
+
+### Overview
+
+Your plugin will source blog posts and authors from the demo API, link the posts and authors, and take image URLs from the posts and optimize them automatically. You'll be able to configure your plugin in your site's `gatsby-config.js` file and write GraphQL queries to access your plugin's data.
+
+The code for this tutorial can be referenced from [its place in the examples folder of the Gatsby repository](https://github.com/gatsbyjs/gatsby/tree/master/examples/creating-source-plugins). You can clone this code and delete the `source-plugin` and `example-site` folders to follow along.
+
+#### An example API request
+
+To see the API in action you can run it locally by navigating into the `api` folder, installing dependencies with `npm install`, and starting the server with `npm start`. You will then be able to navigate to a GraphQL playground running at `http://localhost:4000`. This is a GraphQL server running in Node.js and is _separate from Gatsby_, this server could be replaced with a different backend and the patterns in this tutorial would remain the same.
+
+If you paste the following query into the left side of the window and press the play button you should see data for posts with their IDs and descriptions returned:
+
+```graphql
+query {
+  posts {
+    id
+    description
+  }
+}
+```
+
+This data is an example of the data you will source with your plugin.
+
+_You can also see a running version of the GraphQL playground associated with a distinct API at [https://gatsby-source-plugin-api.glitch.me/](https://gatsby-source-plugin-api.glitch.me/), which is running the `api` folder in a Glitch project, like you would when you run `npm start` on your own computer._
```

Comment:
Could/Should people also use this instead of running it locally?

Reply:
Yes, it is possible to use this endpoint instead. But I opted not to totally encourage that since a shared endpoint with public update access could result in some weird scenarios where someone is updating data and it changes for someone else going through the tutorial at the same time. 

A benefit of Glitch is that it sleeps after a while of inactivity and since the data is stored in-memory, it erases the data back to its original state after a few minutes of not using it.

For me, being able to see that code on the backend was helpful for me in understanding where the plugin sat in the grander scheme of things though 🤷‍♂️